src/iteration.jl

```diff
@@ -2,76 +2,85 @@
 Tables.rowaccess(::Type{<:File}) = true
 Tables.rows(f::File) = f
 
-struct Row{T}
-    file::File
-    row::T
+@inline Base.@propagate_inbounds function Base.getindex(f::File{false}, row::Int)
+    @boundscheck checkbounds(f, row)
+    return Row{false}(getnames(f), getcolumns(f), getlookup(f), row, 0, 0)
 end
 
-getfile(r::Row) = getfield(r, :file)
-getrow(r::Row) = getfield(r, :row)
-
-Base.propertynames(r::Row) = getnames(getfile(r))
+@inline Base.@propagate_inbounds function Base.getindex(f::File{true}, row::Int)
+    @boundscheck checkbounds(f, row)
+    c = getcolumn(f, 1)
```

Comment:
what if `f` has `0` columns?

Reply:
If `f` has 0 columns, then it _has_ to have 0 rows, which get checked in the `checkbounds(f, row)` above.

Reply:
right 👍